docs/index.rst

```diff
@@ -17,6 +21,32 @@ More in-depth coverage is provided in other sections.
     api/index
     examples
 
+
+Concepts
+--------
+
+Besides code which implements Widgetastic and navmazing functionality, two
+basic concepts used to abstract Satellite/Foreman UI functionality are Views
+and Entities.
+
+**View** is part of page containing widgets, it describes appearance of the
+page. Here all the nasty XPaths goes, but please create Foreman or Katello
+pull requests for `id=...` additions where reasonable, as generally XPaths
+are too fragile.
+
+**Entity** is a functional side of application (Satellite or Foreman in our
+case) for some object in it. It defines actions you can do with the object.
+This is the API that end tests uses.
+
+AirGun adds bunch of **widgets** (which are specific to Satellite / Foreman
+web UI) to these defined by Widgetastic (like generic `Text`). Widgets allows
+you to interact with various UI elements and are used in views.
+
+And finally tests. Tests are not part of AirGun repository. You can
+contribute tests to the Robottelo (currently only in abalakh's fork, `branch
```

Comment:
We need to not forget update that information when we move code to proper repo. Do we have anything like #todo tag?

Reply:
I wanted to use this to somehow set expectations - just to express "do not waste your time with python 2" and "do not attempt to run on 6.3". But I agree this tends to become outdated. Any idea?

Ad proper repo: IMO we can fix first moment when somebody complains about it.